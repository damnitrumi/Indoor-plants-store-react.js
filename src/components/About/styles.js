import styled, { css } from "styled-components";
import { Container as SectionWrapper } from "../SectionWrapper/styles";
import { Container as Text } from "../Text/styles";

export const Container = styled.div`
  ${() => css`
    counter-reset: TextCounter;
    ${SectionWrapper} {
      display: flex;
      flex-direction: column;
      gap: 180px;
    }
  `}
`;
export const AboutContainer = styled.div`
  ${({ theme }) => css`
    ${Text}::before {
      content: counter(TextCounter) ".";
      counter-increment: TextCounter;
      font-size: ${theme.font.sizes.large};
      font-family: ${theme.font.family.primary};
      margin-right: ${theme.spacings.small};
      color: ${theme.colors.lightGreen};
    }

    padding: ${theme.spacings.small} 0;
    /* margin-top: 180px; */
    overflow-y: auto;
  `}
`;
