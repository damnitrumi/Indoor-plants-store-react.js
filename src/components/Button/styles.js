import styled, { css } from "styled-components";

export const Container = styled.button`
  ${({ theme }) => css`
    background: transparent;
    border: none;
    cursor: pointer;
    padding: ${theme.spacings.tiny};
    color: white;
    position: relative;
  `}
`;
