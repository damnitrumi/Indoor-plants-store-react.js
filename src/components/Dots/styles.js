import styled, { css } from "styled-components";

export const Container = styled.span`
  ${({ theme, current, index }) => css`
    display: inline-block;
    border-radius: 50%;
    height: ${theme.spacings.tiny};
    width: ${theme.spacings.tiny};
    background-color: ${current == index
      ? theme.colors.grey
      : theme.colors.mediumGrey};
  `}
`;
