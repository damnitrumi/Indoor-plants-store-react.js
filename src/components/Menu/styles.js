import styled, { css } from "styled-components";

export const Container = styled.nav`
  ${() => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
  `}
`;

export const PopUp = styled.span`
  ${() => css`
    display: block;
    position: absolute;
    right: 0;
    background-color: red;
    color: white;
    border-radius: 50%;
    width: 20px;
    height: 20px;
    display: flex;
    justify-content: center;
    align-items: center;
  `}
`;
