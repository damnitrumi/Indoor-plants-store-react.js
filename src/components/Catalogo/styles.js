import styled, { css } from "styled-components";
import { Container as SectionWrapper } from "../SectionWrapper/styles";
import { Container as ImageContainer } from "../ImageContainer/styles";

// export const SectionContainer = styled.div`
//   ${() => css`
//     display: flex;
//     flex-direction: column;
//     height: 100%;
//   `}
// `;

export const Container = styled.div`
  ${({ visible }) => css`
    height: 100vh;
    display: flex;
    flex-direction: column;
    transition: transform 0.5s;
    transform: translateX(-100%);
    position: absolute;
    width: 360px;
    ${visible &&
    css`
      transform: translateX(0);
    `}
    ${SectionWrapper} {
      flex: 1;
      display: flex;
      flex-direction: column;
    }
  `}
`;

export const SlidesContainer = styled.div`
  ${() => css`
    /* ${ImageContainer} {
      height: 460px;
    } */
  `}
`;

export const SliderContainer = styled.div`
  ${() => css`
    flex: 1;
  `}
`;

export const BgContainer = styled.span`
  ${({ theme }) => css`
    display: block;
    background-color: ${theme.colors.lightGrey};
    width: 230px;
    height: 300px;
    border-radius: ${theme.spacings.medium};
    position: absolute;
    z-index: -1;
    top: 40px;
  `}
`;

export const DotsContainer = styled.div`
  ${() => css`
    display: flex;
    justify-content: center;
    gap: 3px;
    position: absolute;
    left: 150px;
    bottom: 290px;
  `}
`;
