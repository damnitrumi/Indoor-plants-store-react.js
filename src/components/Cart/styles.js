import styled, { css } from "styled-components";
import { Container as SectionWrapper } from "../SectionWrapper/styles";
import { Container as Button } from "../Button/styles";
import { Container as Heading } from "../Heading/styles";

export const Container = styled.div`
  ${({ theme, visible }) => css`
    height: 100vh;
    transition: transform 0.5s;
    transform: translateX(100%);
    position: absolute;
    top: 0;
    width: 360px;
    ${visible &&
    css`
      transform: translateX(0);
    `}
    ${SectionWrapper} {
      display: flex;
      flex-direction: column;
      > ${Button} {
        text-align: center;
        background-color: ${theme.colors.lightGreen};
        display: block;
        width: 85%;
        margin: 0 auto;
        border-radius: 5px;
        font-size: ${theme.font.sizes.small};
      }

      > ${Heading} {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const ItensContainer = styled.div`
  ${() => css`
    overflow-y: auto;
    flex: 1;
  `}
`;
