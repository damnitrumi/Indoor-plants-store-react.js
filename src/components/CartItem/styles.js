import styled, { css } from "styled-components";

import { Container as Heading } from "../Heading/styles";
import { Container as Button } from "../Button/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    margin-top: ${theme.spacings.small};
    ${Heading} {
      margin-bottom: 0;
      font-size: 1.4rem;
      font-family: ${theme.font.family.secondary};
    }
    ${Button} {
      color: initial;
    }
  `}
`;

export const ProductImageContainer = styled.div`
  ${() => css`
    flex: 1;
  `}
`;
export const ProductData = styled.div`
  ${({ theme }) => css`
    flex: 2;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding-right: ${theme.spacings.large};
  `}
`;

export const NameContainer = styled.div`
  ${() => css`
    display: flex;
    justify-content: space-between;
  `}
`;

export const ProductOptionsContainer = styled.div`
  ${() => css``}
`;

export const QuantityContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    font-size: ${theme.font.sizes.medium};
  `}
`;
export const Quantity = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    border: 1px solid rgba(0, 0, 0, 0.3);
    ${Button}:first-of-type {
      border-right: 1px solid rgba(0, 0, 0, 0.3);
    }

    ${Button}:last-of-type {
      border-left: 1px solid rgba(0, 0, 0, 0.3);
    }

    span {
      padding: 0 ${theme.spacings.tiny};
    }
  `}
`;
