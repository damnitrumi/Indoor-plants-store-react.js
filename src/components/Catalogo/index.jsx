import * as Styled from "./styles";
import Slider from "react-touch-drag-slider";
import { useState } from "react";
import { SectionWrapper } from "../SectionWrapper";
import { Menu } from "../Menu";
import { ImageContainer } from "../ImageContainer";
import { PlantInfo } from "../PlantInfo";
import { Dots } from "../Dots";
import { useLayoutContext } from "../contexts/LayoutContext";

const plants = [
  {
    name: "Cacto Plant 1",
    price: "20.00",
    details: "Cactus are succulent perennial plants",
    reviews:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc lobortis aliquam elit. Vestibulum eget metus",
    about:
      "Nulla eu massa facilisis, pulvinar nulla at, varius mauris. Fusce id fringilla dolor, ac auctor.",
    image: "/assets/images/plant.png",
    quantity: 0,
  },
  {
    name: "Cacto Plant 2",
    price: "20.00",
    details: "Cactus are succulent perennial plants",
    reviews:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc lobortis aliquam elit. Vestibulum eget metus",
    about:
      "Nulla eu massa facilisis, pulvinar nulla at, varius mauris. Fusce id fringilla dolor, ac auctor.",
    image: "/assets/images/plant.png",
    quantity: 0,
  },
  {
    name: "Cacto Plant 3",
    price: "21.00",
    details: "Cactus are succulent perennial plants",
    reviews:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc lobortis aliquam elit. Vestibulum eget metus",
    about:
      "Nulla eu massa facilisis, pulvinar nulla at, varius mauris. Fusce id fringilla dolor, ac auctor.",
    image: "/assets/images/plant.png",
    quantity: 0,
  },
  {
    name: "Cacto Plant 4",
    price: "22.00",
    details: "Cactus are succulent perennial plants",
    reviews:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc lobortis aliquam elit. Vestibulum eget metus",
    about:
      "Nulla eu massa facilisis, pulvinar nulla at, varius mauris. Fusce id fringilla dolor, ac auctor.",
    image: "/assets/images/plant.png",
    quantity: 0,
  },
  {
    name: "Cacto Plant 5",
    price: "23.00",
    details: "Cactus are succulent perennial plants",
    reviews:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc lobortis aliquam elit. Vestibulum eget metus",
    about:
      "Nulla eu massa facilisis, pulvinar nulla at, varius mauris. Fusce id fringilla dolor, ac auctor.",
    image: "/assets/images/plant.png",
    quantity: 0,
  },
];

export const Catalogo = () => {
  const [current, setCurrent] = useState(0);

  const [, showCatalog] = useLayoutContext();
  return (
    <Styled.Container visible={showCatalog}>
      <SectionWrapper>
        <Menu dark />
        <Styled.SliderContainer>
          <Slider
            onSlideComplete={(i) => {
              console.log("finished dragging, current slide is", i);
              setCurrent(i);
            }}
            onSlideStart={(i) => {
              console.log("started dragging on slide", i);
            }}
            activeIndex={0}
            threshHold={100}
            transition={0.5}
            scaleOnDrag={true}
          >
            {plants.map(({ image }, i) => {
              return (
                <Styled.SlidesContainer key={i}>
                  <ImageContainer urlImg={image} />
                  <Styled.BgContainer />
                </Styled.SlidesContainer>
              );
            })}
          </Slider>
        </Styled.SliderContainer>
      </SectionWrapper>
      <Styled.DotsContainer>
        {plants.map((el, i) => {
          return <Dots current={current} index={i} key={i} />;
        })}
      </Styled.DotsContainer>
      <PlantInfo />
    </Styled.Container>
  );
};
