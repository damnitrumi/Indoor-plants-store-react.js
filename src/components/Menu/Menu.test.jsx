import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { Menu } from ".";
import { Catalogo } from "../Catalogo";

import plants from "../utils/plantsMock";
import { userEvent } from "@storybook/testing-library";

describe("<Menu />", () => {
  it("should render Menu and its components", () => {
    renderTheme(<Menu />);
    expect(screen.getAllByRole("img")).toHaveLength(3);
  });

  it("should render Menu and its components with pop up", () => {
    renderTheme(
      <>
        <Menu /> <Catalogo plants={plants} />
      </>
    );

    const catalogBtn = screen.getByRole("button", { name: "Add To Cart" });
    userEvent.click(catalogBtn);
    expect(screen.getAllByLabelText("ItemsQty")).toHaveLength(2);
  });

  it("should match snapshot", () => {
    const { container } = renderTheme(<Menu />);
    expect(container).toMatchSnapshot();
  });
});
