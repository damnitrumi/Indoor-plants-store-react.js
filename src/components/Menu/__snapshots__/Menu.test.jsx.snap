// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0.8rem;
  color: white;
  position: relative;
}

<div>
  <nav
    class="c0"
  >
    <button
      class="c1"
    >
      <img
        class=""
        src="/assets/images/menu-light.png"
      />
    </button>
    <button
      class="c1"
    >
      <img
        class=""
        src="/assets/images/short-logo-light.png"
      />
    </button>
    <button
      class="c1"
    >
      <img
        class=""
        src="/assets/images/cart-light.png"
      />
    </button>
  </nav>
</div>
`;
